Dockerfile.ubi8

```diff
@@ -10,17 +20,24 @@ COPY go.sum go.sum
 # and so that source changes don't invalidate our downloaded layer
 # RUN go mod download
 
+COPY Makefile Makefile
+COPY Makefile.helm.mk Makefile.helm.mk
+COPY Makefile.helper.mk Makefile.helper.mk
+COPY Makefile.specialresource.mk Makefile.specialresource.mk
+COPY hack/ hack/
+
 # Copy the go source
 COPY main.go main.go
 COPY api/ api/
 COPY cmd/ cmd/
 COPY controllers/ controllers/
+COPY .patches/ .patches/
 
 COPY pkg/ pkg/
 COPY vendor/ vendor/
 
 # Build
-RUN CGO_ENABLED=0 GO111MODULE=on go build -mod=vendor -a -o manager main.go
+RUN ["make", "manager"]
```

Comment:
I think we need `patch` as a dependency of several targets? At least `lint` and `manager`.

Reply:
What do you think about this solution instead https://github.com/openshift-psap/special-resource-operator/pull/172?
@qbarrand @pacevedom 

This is coming with the assumption that we don't use `src` image in Prow unless we actually want to deploy SRO (but not for lifting or UT for example).

Reply:
Not sure I understand your first comment @ybettan, what do you mean? We need the dependency for other targets as @qbarrand says.

I agree that #172 scripted solution looks cleaner than having it in `Makefile`.

As for the `src` image, it's basically the base image (`build_root`) with the code already cloned. This will have all the needed dependencies there, including patch.

Reply:
I agree that `local-image-build` should probably not require the `patch` target itself, if we use `make manager` in the `Dockerfile`.

Reply:
> Not sure I understand your first comment @ybettan, what do you mean? We need the dependency for other targets as @qbarrand says.

Agree, I was mistaken on that one.

> As for the src image, it's basically the base image (build_root) with the code already cloned. This will have all the needed dependencies there, including patch.

Regarding this one I will open a slack thread for this discussion.